pkg/virt-launcher/virtwrap/converter/compute/graphics.go

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package compute
+
+import (
+	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type GraphicsDomainConfigurator struct {
+	architecture      string
+	bochsForEFIGuests bool
+}
+
+type graphicsOption func(*GraphicsDomainConfigurator)
+
+func NewGraphicsDomainConfigurator(options ...graphicsOption) GraphicsDomainConfigurator {
+	var configurator GraphicsDomainConfigurator
+
+	for _, f := range options {
+		f(&configurator)
+	}
+
+	return configurator
+}
+
+func (g GraphicsDomainConfigurator) Configure(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
+	if vmi.Spec.Domain.Devices.AutoattachGraphicsDevice != nil && !*vmi.Spec.Domain.Devices.AutoattachGraphicsDevice {
+		return nil
+	}
+
+	g.configureVNCSocket(vmi, domain)
+
+	return nil
+}
+
+func (g GraphicsDomainConfigurator) configureVNCSocket(vmi *v1.VirtualMachineInstance, domain *api.Domain) {
+	domain.Spec.Devices.Graphics = []api.Graphics{
```

Comment:
Does the order of elements defined in `domain.Spec.Devices.Graphics` change following this commit?